pkg/kubectl/resource/builder.go

```diff
@@ -655,7 +655,7 @@ func (b *Builder) visitByResource() *Result {
 			selectorNamespace = ""
 		} else {
 			if len(b.namespace) == 0 {
-				return &Result{singular: isSingular, err: fmt.Errorf("namespace may not be empty when retrieving a resource by name")}
+				return &Result{singular: isSingular, err: fmt.Errorf("all namespaces or empty namespace is invalid when retrieving a resource by name")}
```

Comment:
I'd prefer we keep track of whether AllNamespaces was called so we can give a more specific error message in that case